src/java.base/share/classes/java/security/SecureRandom.java

```diff
@@ -267,23 +275,34 @@ public SecureRandom(byte[] seed) {
     private void getDefaultPRNG(boolean setSeed, byte[] seed) {
         Service prngService = null;
         String prngAlgorithm = null;
-        for (Provider p : Providers.getProviderList().providers()) {
-            // SUN provider uses the SunEntries.DEF_SECURE_RANDOM_ALGO
-            // as the default SecureRandom algorithm; for other providers,
-            // Provider.getDefaultSecureRandom() will use the 1st
-            // registered SecureRandom algorithm
-            if (p.getName().equals("SUN")) {
-                prngAlgorithm = SunEntries.DEF_SECURE_RANDOM_ALGO;
+
+        // If in FIPS mode, use the SecureRandom from the FIPS provider
```

Comment:
Please finish the sentence with a period.

Reply:
added.